### DISCUSSION TITLE
NUMBER_OF_TYPE_ARGUMENTS_MISMATCH

### DISCUSSION THREAD

**Discussion by Pelumi527** [aptos-labs/aptos-developer-discussions#80] [Questions]
I am trying to call this function
```
{
  "name": "create_user",
  "visibility": "public",
  "is_entry": true,
  "is_view": false,
  "generic_type_params": [
    {
      "constraints": []
    }
  ],
  "params": [
    "&signer",
    "address",
    "0x1::string::String",
    "0x1::option::Option<address>",
    "0x1::string::String"
  ],
  "return": []
}
```
and the payload looks like this

```
const transaction = {
    type: "entry_function_payload",
    function: `${0x1}::core::create_user`,
    type_arguments: [],
    arguments: [
    "0xf657dd8a6f5fb6da917d95c70f53244a424461a89b3f1fc7e193874d01cb3457",
      "1111",
      undefined,
      "pelumi"
      ],
  };
```

**Comment by Alexzander-Stone** [comment]
Hi @Pelumi527, could you add a bit more context for me? It'll help in assisting you.

1. What are you expecting to happen vs what is actually happening?
2. Can you link your JS/TS code as a reference? Include how the `transaction` object is used.

**Reply by Pelumi527** [reply]
Thanks, for your help figured out the issue

**Comment by Pelumi527** [comment]
> Hi @Pelumi527, could you add a bit more context for me? It'll help in assisting you.
> 
> 1. What are you expecting to happen vs what is actually happening?
> 2. Can you link your JS/TS code as a reference? Include how the `transaction` object is used.
